trie/trie.go

```diff
@@ -1031,15 +1039,15 @@ func (t *Trie) unload(hex []byte, h *hasher) {
 		i1, i2 := p.childrenIdx()
 		switch hex[len(hex)-1] {
 		case i1:
-			t.unloadFunc(hex, p.child1.hash())
```

Comment:
I just noticed now that `unloadNodeFunc` is only called for nodes just above the top of the unloaded trie. Which missed all the intermediate nodes below. This is because `trie_pruning.go` tries to minimise the invocations of function `unload`:
````
var empty = false
	sort.Strings(hexes)
	for i, hex := range hexes {
		if i == 0 || len(hex) == 0 || !strings.HasPrefix(hex, hexes[i-1]) { // If the parent nodes are pruned, there is no need to prune descendants
			t.unload([]byte(hex))
			if len(hex) == 0 {
				empty = true
			}
		}
	}
````

Reply:
- If you mean "we can put to hashBucket all hashNodes from unloaded sub-tree" - then probably we already did put it to hashBucket when that hashNodes were created.

- if you mean "we can calculate hashes for all nodes in unloaded sub-tree and save it to hashBucket" - yes, maybe it's good idea.

- if you mean something different, please share.

Reply:
Did you mean to has all nodes in subtree? Like here: https://github.com/ledgerwatch/turbo-geth/pull/355/commits/83cb14fd4e067de0b997007122718388d14b8462#diff-a18deccdc0311173b4a50bcd4c881586R151

Reply:
Yes, that's what I meant. Though you might not need to calculate the hashes, but just use node.hash() function. Assuming this is happening after the block is processed and the trie.Hash() function has been called, everything should be consistent